lib/mayaUsd/ufe/UsdShaderNodeDef.cpp

```diff
@@ -109,6 +109,29 @@ std::string UsdShaderNodeDef::type() const
     return fShaderNodeDef->GetIdentifier();
 }
 
+namespace {
+// clang-format off
+TF_DEFINE_PRIVATE_TOKENS(
+    _tokens,
+
+    (arnold)
+    (shader)
+);
+// clang-format on
+
+// Arnold nodes seem to be incompletely registered, this affects the "classification" scheme used
+// by the UFE abstraction. This has been identified as Arnold-USD issue 1214:
+//    https://github.com/Autodesk/arnold-usd/issues/1214
+// We will detect this in a way that should switch back to the normal classification scheme if the
+// registration code is updated.
+bool _isArnoldWithIssue1214(const PXR_NS::SdrShaderNode& shaderNodeDef)
+{
+    return shaderNodeDef.GetSourceType() == _tokens->arnold
+        && shaderNodeDef.GetFamily() == _tokens->shader
+        && shaderNodeDef.GetName() != _tokens->shader;
+}
+} // namespace
+
```

Comment:
No unit test for this one. Would require mtoa to be present (but not necessarily loaded).